### PR TITLE
SOF-7357: add RestAPI class + add browser methods

### PR DESCRIPTION
**Pull request by k0stik**

### CODE REVIEW

File: src/js/cypress/utils/api.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    [key: string]: string | undefined;
+}
+
+export interface Headers {
+    [key: string]: string;
+}
+
+export type Body = object | undefined;
+
+export default class RestApi {
```
Comment (timurbazhirov) [inline src/js/cypress/utils/api.ts]: API - capital

**Review by timurbazhirov — APPROVED** [review]

File: src/js/cypress/Browser.ts [hunk 2]
```diff
@@ -436,18 +456,31 @@ export class IframeBrowser extends Browser {
...
     get(selector: GetParams[0], options?: GetParams[1]) {
-        return this.getIframeBody().find(selector, options);
+        /**
+         * Chaining after find() may lead to the error "cy...() failed because the page updated" in some cases
+         * @see https://docs.cypress.io/guides/references/error-messages#cy-failed-because-the-page-updated
+         * hence .as("iframe_element") and cy.get("@iframe_element")
+         */
+        this.getIframeBody().find(selector, options).as("iframe_element");
+
+        return cy.get("@iframe_element");
```
Comment (timurbazhirov) [inline src/js/cypress/Browser.ts]: Let's isolate the part that uses `@iframe_element` into a separate function b/c it's used here and below

File: src/js/cypress/utils/api.ts (hunk 1 above)
Comment (k0stik) [reply src/js/cypress/utils/api.ts]: Done

File: src/js/cypress/Browser.ts (hunk 2 above)
Comment (k0stik) [reply src/js/cypress/Browser.ts]: Removed this as it didn't work as expected.
Instead added the `assertIframeNotExistWithRetry` method

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**